### ISSUE TITLE
The performance of my pretraining model is very bad.

### ISSUE THREAD

**Issue by dong-y21** [mkshing/e4t-diffusion#15]
CUDA_VISIBLE_DEVICES=0,1,2,3 accelerate launch pretrain_e4t.py \
  --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" \
  --clip_model_name_or_path="openai/clip-vit-large-patch14" \
  --domain_class_token="face" \
  --placeholder_token="*s" \
  --prompt_template="a photo of {placeholder_token}" \
  --reg_lambda=0.01 \
  --output_dir="pretrained-face-2" \
  --train_image_dataset="FFHQ" \
  --iterable_dataset \
  --resolution=512 \
  --train_batch_size=4 \
  --learning_rate=1.6e-5 \
  --max_train_steps=30000 \
  --mixed_precision="no" \
  --enable_xformers_memory_efficient_attention 

1. The sampled images of monet style almost have no human face. 
  10k steps:
![media_images_train_samples_9999_95b0ef88ec5a15963598](https://user-images.githubusercontent.com/106572559/228279251-301667a2-2894-46e9-812c-f902cd76132c.png)
 and sampled results become weird after 13k steps like this:
![media_images_train_samples_24999_6f71f54ac22c2abbf3ad (1)](https://user-images.githubusercontent.com/106572559/228280350-49b58041-ed23-4b22-9bad-045f1c85d44c.png)

2. If I use 10k pretrained model for finetuning, I get totally unrelated faces by inference. (*s on the beach)
<img width="1021" alt="image" src="https://user-images.githubusercontent.com/106572559/228283243-1fd87f10-b3c1-4820-9d96-83ed6f8e06e1.png">

If I use 20k or 30k pretrained model for finetuning, I get totally weird images without any face. (*s on the beach)
<img width="1020" alt="image" src="https://user-images.githubusercontent.com/106572559/228282798-8ccc86f5-7176-49f6-900a-b903da8600db.png">

So do I have to train the model with a mixed dataset of FFHQ and CelebA-HQ instead of using only FFHQ dataset? What can I do to get better results?

**Comment by Mikhail-off** [comment]
Does loss_reg equal 0?

**Comment by dong-y21** [comment]
> Does loss_reg equal 0?

The loss seems reasonable.
<img width="1652" alt="image" src="https://user-images.githubusercontent.com/106572559/228293318-74892f70-a0d8-4c38-b2df-6da89abb6e02.png">

**Comment by Mikhail-off** [comment]
https://github.com/mkshing/e4t-diffusion/blob/main/pretrain_e4t.py
427, 429, 445

you have loss_reg=args.reg_lambda(=0.01) * domain_embed.pow(2).sum() = 1e-4 => domain_embed.pow(2).sum() = 0.01. I think its very low to make sense. And its main information about the input image. 

I have same problems. Changed reg_lambda from 1e-2 to 1e-4 few hours ago

**Comment by dong-y21** [comment]
> https://github.com/mkshing/e4t-diffusion/blob/main/pretrain_e4t.py 427, 429, 445
> 
> you have loss_reg=args.reg_lambda(=0.01) * domain_embed.pow(2).sum() = 1e-4 => domain_embed.pow(2).sum() = 0.01. I think its very low to make sense. And its main information about the input image.
> 
> I have same problems. Changed reg_lambda from 1e-2 to 1e-4 few hours ago

Thank you, I will try this.

**Comment by Mikhail-off** [comment]
@dong-y21 any results?)

**Comment by dong-y21** [comment]
> @dong-y21 any results?)

Still very few faces appear in monet style

**Comment by mkshing** [comment]
I changed a lot of parts to get better results and released the pre-trained model for face. You should check the latest code and reopen this issue if you have still problems.
